<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3> Analyst: Maricel Maribojoc</h3>

In [1]:
#import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Converting Dataset(s) into DataFrame</h3>
<h2>Requirements</h2>
<ul>
    <li>
        Name of the DataFrame
    </li>
    <li>
        File Path
    </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [7]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [8]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [9]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [10]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [11]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [12]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [13]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [14]:
penguins[penguins['body_mass_g'] >= 400]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
4       Adelie  Torgersen            36.7           19.3              193.0   
5       Adelie  Torgersen            39.3           20.6              190.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
5         3650.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[342 rows x 8 columns]

In [15]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [16]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [17]:
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h2>FILLING UP NULL</h2>

In [18]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [19]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [20]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [22]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [23]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [24]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [25]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [26]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [27]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [28]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [29]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [30]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [31]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [32]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [33]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


<h2>Changing the Data</h2>

In [35]:
penguins['species'] = penguins['species'].astype('category')

In [36]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
339  4000.000000    male  2009  
340  3400.000000  female  2009  
341  3775.000000    male  2009  
342  4100.000000    male  2009  
343  3775.000000  female  2009  

[344 rows x 8 columns]

<h2>Creating a Column</h2>

In [37]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [38]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000 , 1)

In [39]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [40]:
penguins['mass_kg'].max()

np.float64(6.3)

In [41]:
penguins['mass_kg'] == 6.3

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: mass_kg, Length: 344, dtype: bool

In [42]:
penguins[penguins['mass_kg'] == 6.3]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  mass_kg  
169       6300.0  male  2007      6.3

In [43]:
penguins[penguins['mass_kg'] == 4.2]['island']

3      Torgersen
9      Torgersen
19     Torgersen
35         Dream
49         Dream
73     Torgersen
75     Torgersen
83     Torgersen
103       Biscoe
139        Dream
146        Dream
166       Biscoe
168       Biscoe
198       Biscoe
212       Biscoe
271       Biscoe
284        Dream
301        Dream
Name: island, dtype: object

<h1>Categorical Data</h1>
<hr>
<h3>body mass v sex</h3>

In [44]:
#import scipy.stats
from scipy.stats import f_oneway

In [45]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [46]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [47]:
stats, p_value = f_oneway (penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] =='female']['body_mass_g'])

In [48]:
p_value

np.float64(3.0226600487050246e-15)

<h2>Explanatory Data Analytics (EDA)</h2>
<hr>

<h2>Question 1</h2>
<h3>How many female species recorded under Gentoo?</h3>
<hr>

In [49]:
((penguins['species'] == 'Gentoo') & (penguins['sex'] == 'female')).sum()

np.int64(58)

<h2>Insight 1</h2>
<p>Based on the result above, there are a total of 58 female penguins in Gentoo.</p>
<hr>

<h2>Question 2</h2>
<h3>In the given data, what penguin sex has recorded the most?</h3>
<hr>

In [50]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h2>Insight 2</h2>
<p>Among the three penguin species, the most recorded sex was male that has 179.</p>
<hr>

<h2>Question 3</h2>
<h3>Which penguin species has the longest average bill length, and what might this indicate about its feeding habits compared to other species?</h3>
<hr>

In [51]:
adelie_df = np.round(penguins[penguins['species'] == 'Adelie']['bill_length_mm'].mean(), 1)

In [52]:
adelie_df

np.float64(38.8)

In [53]:
chinstrap_df = np.round(penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'].mean(), 1)

In [54]:
chinstrap_df

np.float64(48.8)

In [55]:
gentoo_df = np.round(penguins[penguins['species'] == 'Gentoo']['bill_length_mm'].mean(), 1)

In [56]:
gentoo_df

np.float64(47.5)

<h2>Insight 3</h2>
<p>Based on the dataset, Chinstrap penguins have the longest bill length (48.8 mm), followed closely by Gentoo penguins (47.5 mm), while Adelie penguins have the shortest bills (38.8 mm). This variation in bill length likely reflects differences in their feeding strategies and ecological niches, with longer bills better suited for catching larger or deeper-dwelling prey, and shorter bills adapted for feeding on surface-level organisms.</p>
<hr>

<h2>Question 4</h2>
<h3>How many penguins have a bill depth less than 15 mm and flipper length greater than 200 mm?</h3>
<hr>

In [57]:
(penguins['bill_depth_mm'] < 15).sum()

np.int64(60)

In [58]:
(penguins['flipper_length_mm'] > 200).sum()

np.int64(150)

<h2>Insight 4</h2>
<p>There are a total of 60 penguins that has a bill length of less than 15 mm and a total of 150 penguins that has a greater than 200 mm flipper length.</p>
<hr>

<h2>Question 5</h2>
<h3>How many female penguins are there in Chinstrap?</h3>
<hr>

In [59]:
((penguins['species'] == 'Chinstrap') & (penguins['sex'] == 'female')).sum()

np.int64(34)

<h2>Insight 5</h2>
<p>There are only 34 female belongs to Chinstrap species.</p>
<hr>

<h2>Question 6</h2>
<h3>In what year has the body mass of 5.5 kg?</h3>
<hr>

In [60]:
penguins[penguins['mass_kg'] == 5.5]['year']

255    2009
257    2009
259    2009
261    2009
265    2009
Name: year, dtype: int64

<h2>Insight 6</h2>
<p>In the year of 2009, penguins weighted 5.5 kg.</p>
<hr>

<h2>Question 7</h2>
<h3>Compare the average body mass of male vs. female penguins.</h3>
<hr>

In [61]:
penguins[penguins['sex'] == 'male']['body_mass_g'].mean()

np.float64(4514.684406547094)

In [62]:
penguins[penguins['sex'] == 'female']['body_mass_g'].mean()

np.float64(3862.2727272727275)

<h2>Insight 7</h2>
<p>Male penguins tend to have a higher average body mass than female penguins.</p>
<hr>

<h2>Question 8</h2>
<h3>What is the average flipper length and bill length for each species?</h3>
<hr>

In [63]:
penguins.groupby('species')[['flipper_length_mm', 'bill_length_mm']].mean()

flipper_length_mm  bill_length_mm
species                                     
Adelie            190.025758       38.825144
Chinstrap         195.823529       48.833824
Gentoo            217.055768       47.475983

<h2>Insight 8</h2>
<p>In terms of flipper length, Gentoo penguins tends to have the longest flippers compared to Chintraps and Adelie. When it comes to bill length Chinstrap has the longest bill than the other two species. Adelie has the shorter bill and flipper compared to Gentoo and Chinstrap.</p>

<hr>

<h2>Question 9</h2>
<h3>What is the average body mass for male and female penguins of each species?</h3>
<hr>


In [64]:
avg_body_mass_by_sex_species = penguins.groupby(['species', 'sex'])['body_mass_g'].mean()

In [65]:
avg_body_mass_by_sex_species

species    sex   
Adelie     female    3368.835616
           male      4013.629802
Chinstrap  female    3527.205882
           male      3938.970588
Gentoo     female    4679.741379
           male      5411.011430
Name: body_mass_g, dtype: float64

<h2>Insight 9</h2>
<p>The analysis above helps to understand sexual dimorphism within each species regarding body mass. It's common for male and female penguins of the same species to have different average body masses, often with males being larger.</p>
<hr>

<h2>Question 10</h2>
<h3>What is the correlation between bill length and flipper length? </h3>
<hr>

In [66]:
corr_bill_flipper = penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])

In [67]:
corr_bill_flipper

np.float64(0.6561813407464278)

<h2>Insight 10</h2>
<p>The correlationn of bill lenght to flipper lenght is 0.66.</p>
<hr>

<h2>Question 11</h2>
<h3>What is the median bill length for each species?</h3>
<hr>

In [68]:
med_bill_length = penguins.groupby('species')['bill_length_mm'].median()

In [69]:
med_bill_length 

species
Adelie       38.80
Chinstrap    49.55
Gentoo       47.25
Name: bill_length_mm, dtype: float64

<h2>Insight 11</h2>
<p>In terms of median bill length, Chinstrap bill length differ compared to Adelie and Gentoo.</p>
<hr>

<h2>Question 12</h2>
<h3>What is the tallest recorded flipper length in the dataset?</h3>
<hr>

In [70]:
flipper_length = (penguins.groupby('species')['flipper_length_mm'].max())

In [71]:
flipper_length

species
Adelie       210.0
Chinstrap    212.0
Gentoo       231.0
Name: flipper_length_mm, dtype: float64

<h2>Insight 12</h2>
<p>Based on the analysis above, the Gentoo has the tallest flipper length in the dataset, which is 231.</p>
<hr>

<h2>Question 13</h2>
<h3>How many penguins are recorded per year?</h3>
<hr>

In [72]:
penguins_per_year = penguins['year'].value_counts()

In [73]:
penguins_per_year

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h2>Insight 13</h2>
<p>In the year 2009, there are 120 penguins that are being recorded. The year 2008 has 114 penguins, and 110 penguins were recorded in the year 2007.</p>